### PR TITLE
GenAi: add a new cron job to delete old Aichat data 

### PR DESCRIPTION
**Pull request by Erin007**
[LABS-962](https://codedotorg.atlassian.net/browse/LABS-962)

This sets up a cron job once daily at 8pm pacific to delete any `AichatEvent` or `AichatRequest` older than 90 days. It also sends a little messages to the "#cron-daily" Slack channel so it's easy to keep an eye on the process and volume of deletions. 

There's follow up work to fully delete the unused `AichatSessions` table here: [LABS-976](https://codedotorg.atlassian.net/browse/LABS-976)

### CODE REVIEW

File: bin/cron/delete_old_ai_chat_data [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+require_relative '../../dashboard/config/environment'
+
+# We want to avoid storing student chat messages any longer than
+# is necessary for analysis to prevent storing excess student PII. This deletes
+# any AichatEvent or AichatRequest older than 90 days.
+# 
+# Additionally, we delete any AichatSession older than 30 days. AichatSession is
+# no longer in use and we plan to fully delete the table.
+# See https://codedotorg.atlassian.net/browse/LABS-976 for details.
```
Comment (sanchitmalhotra126) [inline bin/cron/delete_old_ai_chat_data]: Oh oops I think I misspoke over Slack - I think 90 days (same as the other two) should also work for AichatSessions for now? That way we still have a bit of time to review any last minute data from our July pilot if necessary

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

And thanks for the detailed follow-up task!

File: bin/cron/delete_old_ai_chat_data [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Log the number of records deleted for each table to Slack, #cron-watcher-aichat-deletion
+# so we can easily monitor the deletion process.
+def main
+  thirty_days_ago = Time.now - 30.days
+  ninety_days_ago = Time.now - 90.days
+
+  destroyed_sessions = AichatSession.where('created_at < ?', thirty_days_ago).destroy_all
```
Comment (fisher-alice) [inline bin/cron/delete_old_ai_chat_data]: Could we hold off on deleting data from this table for now? Asked @samantha-code about this [over Slack](https://codedotorg.slack.com/archives/C06UT1E2796/p1723659639734519) and she would like to keep the student chat data for now.

**Review by fisher-alice — APPROVED** [review]
Thanks for taking care of this! 🎉 🧹 
Just left one comment based on consulting with product.

**Comment by cat5inthecradle** [comment]
Any reason not to use #cron-daily?

**Comment by Erin007** [comment]
> Any reason not to use #cron-daily?

because I did not know it existed 🤣 updating...

File: bin/cron/delete_old_ai_chat_data (hunk 2 above)
Comment (Erin007) [reply bin/cron/delete_old_ai_chat_data]: @fisher-alice @sanchitmalhotra126 I updated to keep student chat data from the pilot, and updated https://codedotorg.atlassian.net/browse/LABS-976 with a reminder to revisit this. Presumably when Product is ready to delete the pilot data, we can both update the cron job and delete `AiSessions`. How does that sound?

**Comment by Erin007** [comment]
Set up a reminder in the generative-ai-unit-eng Slack channel is 90ish days to double check that the deletion is as expected. 
Screenshot 2024-08-18 at 9 48 25 PM

File: bin/cron/delete_old_ai_chat_data (hunk 2 above)
Comment (fisher-alice) [reply bin/cron/delete_old_ai_chat_data]: SGTM! Thanks so much.

**Pull request merged**